### PR TITLE
vscode: improve desktop file handling

### PR DESCRIPTION
**Pull request by eclairevoyant**
## Description of changes

- **vscode: use copyDesktopItems hook instead of manually copying**
- **vscode: remove folder and text mimetype associations**
  - See https://github.com/microsoft/vscode/pull/209510

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/editors/vscode/generic.nix [hunk 1]
```diff
@@ -134,7 +134,7 @@ in
       startupNotify = true;
       startupWMClass = shortName;
       categories = [ "Utility" "TextEditor" "Development" "IDE" ];
-      mimeTypes = [ "text/plain" "inode/directory" ];
+      mimeTypes = [ "text/plain" ];
```
Comment (JohnRTitor) [inline pkgs/applications/editors/vscode/generic.nix:137]: Why keep the `text/plain` association?

Comment (eclairevoyant) [reply pkgs/applications/editors/vscode/generic.nix:137]: vscode is a text editor?

Comment (eclairevoyant) [reply pkgs/applications/editors/vscode/generic.nix:137]: oh, I guess they removed that upstream too.

File: pkgs/applications/editors/vscode/generic.nix
Comment (JohnRTitor) [inline pkgs/applications/editors/vscode/generic.nix:1]: This is still relevant. https://github.com/telamon/vscode/blob/93e65595bdf839c6eb8ae27d484d65662129ae6a/resources/linux/code.desktop#L11

**Review by drupol — APPROVED** [review]

File: pkgs/applications/editors/vscode/generic.nix
Comment (eclairevoyant) [reply pkgs/applications/editors/vscode/generic.nix:1]: Why didn't we just install the upstream file?

Comment (eclairevoyant) [reply pkgs/applications/editors/vscode/generic.nix:1]: Okay that seems to also require installing this file (https://github.com/microsoft/vscode/blob/main/resources/linux/code-workspace.xml) to some location and then referring to it somehow. You're welcome to investigate it in a separate PR, I'd rather not an invalid MIME type here.

**Comment by eclairevoyant** [comment]
vscode seems to hang during the tests, but that appears to be the case even on `master` as well.

**Review by JohnRTitor — APPROVED** [review]

**Comment by JohnRTitor** [comment]
Ran vscode and checked the desktop files.

**Pull request merged**